exporter/kafkaexporter/marshaller.go

```diff
@@ -16,21 +16,30 @@ package kafkaexporter
 
 import (
 	"go.opentelemetry.io/collector/consumer/pdata"
-	otlptrace "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/collector/trace/v1"
 )
 
-// marshaller encodes traces into a byte array to be sent to Kafka.
-type marshaller interface {
-	// Marshal serializes spans into byte array
-	Marshal(traces pdata.Traces) ([]byte, error)
+var marshallers = map[string]Marshaller{}
+
+// GetMarshaller gets a Marshaller for encoding or nil if no marshaller is registered.
+func GetMarshaller(encoding string) Marshaller {
```

Comment:
Does this need to be public?

Reply:
I want to make this extensible for people extending the collector. Anybody who imports a custom format (e.g. in exporter) can easily support Kafka transport.

Reply:
Let's not do this prematurely. If there is a request we will consider to do then. Any public API is extra overhead to maintain

Reply:
We can do that, I wanted to make it simpler to provide custom encodings. The base idea was that the registry could be called from any place e.g. exporter that imports custom data model.

Reply:
Done, I have exposed (un)marshalers in the factory.